canonicalwebteam/blog/feeds.py

```diff
@@ -0,0 +1,46 @@
+from feedgen.entry import FeedEntry
+from feedgen.feed import FeedGenerator
+
+
+def build_feed(uri, title, subtitle, articles, article_url_generator):
+    feed = FeedGenerator()
+    feed.generator("Python Feedgen")
+    feed.id(uri)
+    feed.title(title)
+    feed.link(href=uri, rel="self")
+    feed.description(f"{title} feeds")
+
+    for article in articles:
+        feed.add_entry(build_entry(article, uri, article_url_generator))
+
+    return feed
+
+
+def build_entry(article, uri, article_url_generator):
+    title = article["title"]["rendered"]
+    author = article["_embedded"]["author"][0]
+    description = article["excerpt"]["rendered"]
+    content = article["content"]["rendered"]
+    published = f'{article["date_gmt"]} GMT'
+    updated = f'{article["modified_gmt"]} GMT'
+
+    # TODO: Improve URL generation
+    host = uri.split("/")[2]
+    path = article_url_generator(slug=article["slug"])
```

Comment:
Couldn't we just do `path = article['slug']`? I'm really not sure passing a `article_url_generator` function through 2 levels of encapsulation just to resolve to `/{slug}` is worth it.

In general, I'm not a fan of `url_for` patterns, as in almost every case anyone reading that line just wants to know what the eventual URL looks like, and working that out from a `url_for` function always involves multiple steps. I'm not sure I've ever seen anyone benefit from the theoretical upside of being able to change the URL in just one place (urls.py).

Reply:
Or is this needed because you don't know the subpath? (`/blog` or not)